### DISCUSSION TITLE
Never seen before: Function call from class directly from outside a method (spark:Label) (CodeStyle?)

### DISCUSSION THREAD

**Discussion by 2jfw** [airsdk/Adobe-Runtime-Support#2501] [Q&A]
I came across a part of code at the beginning of the `spark:Label` and found a rather interesting and new-to-me way to call a function.
Have a look at this, where the function `initClass` is called _directly_ from outside any method:

```
public class Label extends TextBase
{
    include "../core/Version.as";

    //--------------------------------------------------------------------------
    //
    //  Class initialization
    //
    //--------------------------------------------------------------------------
    
    /**
     *  @private
     */
    private static function initClass():void
    {
        staticTextBlock = new TextBlock();
        
        staticTextElement = new TextElement();
        
        staticSpaceJustifier = new SpaceJustifier();
        
        staticEastAsianJustifier = new EastAsianJustifier();
        
        if ("recreateTextLine" in staticTextBlock)
            recreateTextLine = staticTextBlock["recreateTextLine"];
    }
    
    initClass();

   [...]
```

**@ajwfrost:** Any feedback on this? Are there any advantages or is it just clean-code/good practice to initialize statics like this?
I've never seen this construct before... 🤔

**Link to Class:**
https://github.com/apache/flex-sdk/blob/develop/frameworks/projects/spark/src/spark/components/Label.as

**Comment by ajwfrost** [comment]
That's an odd one! I've never seen usage like that, I would normally do something like
```
private static var bInit : Boolean = initClass();
```
to trigger the call to a function during the class constructor.

I'd always thought the Java `static { ... }` block was quite a good idea. This one is a bit similar I guess...

Internally, there's a "class initialisation" function where things like the initial values of static variables are set up, so I guess this `initClass();` call is just added to that. But that's a bit of an assumption..!

thanks

**Comment by teotigraphix** [comment]
I remember working on the compiler and ` initClass()` could be;

```
{
     initClass();
}
```

So AS3 as of now actually has a class open scope that can execute code top down. @2jfw 's code could have the braces and with `initClass()` would semantically be the same thing and does work both ways, the parser just sees the new block scope.
